File: src/NuFridge.Website.MVC/App/viewmodels/databinding/NuFridgeInstall.js
define(['knockoutvalidation'], function (validation) {

    ko.validation.init({
        registerExtenders: true,
        insertMessages: false,
    });

    window.NuFridgeInstall = function (config) {
        var self = this           

        var obj = {
            SupportedDatabaseSystems: ko.observableArray(['Mongo', 'Sql Compact']).extend({ required: true }),
            DatabaseSystem: ko.observable('Mongo').extend({ required: true }),  
            MongoDBServer: ko.observable(""),
            MongoDBDatabase: ko.observable(""),
            IISWebsiteName: ko.observable("").extend({ required: true }),
            PortNumber: ko.observable("80").extend({
                required: true,
                pattern: { message: 'Port number must be numeric', params: /^\d+$/ }
            }),
            PhysicalDirectory: ko.observable("").extend({
                required: true,
                pattern: { message: 'Please provide a valid directory path', params: /^([A-Za-z]:)(\\[A-Za-z_\-\s0-9\.]+)+$/ }
            })
        }

        var data = $.extend(obj, config);
        ko.mapping.fromJS(data, {}, self);

        self.MongoDBServer.extend({
            required: {
                onlyIf: function () {
                    return self.DatabaseSystem() == "Mongo";
                }
            }
        });
       
        self.MongoDBDatabase.extend({
            required: {
                onlyIf: function () {
                    return self.DatabaseSystem() == "Mongo";
                }
            }
        });   
       
    };

    window.NuFridgeInstall.mapping = {
        create: function (options) {
            var fd = new NuFridgeInstall(options.data);
            return fd;
        }

    };
});

File: src/NuFridge.Website.MVC/Models/NuFridgeInstall.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace NuFridge.Website.MVC.Models
{


    public class NuFridgeInstall
    {      

        /// <summary>
        /// The database system to use - SQL compact, Mongo etc.
        /// </summary>
        public string DatabaseSystem { get; set; }

        public string MongoDBServer { get; set; }
        public string MongoDBDatabase { get; set; }

        public int PortNumber { get; set; }
        public string IISWebsiteName { get; set; }
        public string PhysicalDirectory { get; set; }
    }
}